ss/pebbledb/db.go

```diff
@@ -267,6 +282,25 @@ func (db *Database) Prune(version int64) error {
 			return fmt.Errorf("invalid MVCC key")
 		}
 
+		storeKey, err := parseStoreKey(currKey)
+
+		// Should never happen given we skip the metadata keys
+		if err != nil {
+			return err
+		}
+
+		// For every new module visited, check to see last time it was updated
+		if storeKey != prevStore {
+			prevStore = storeKey
+			updated, ok := db.storeKeyDirty.Load(storeKey)
+			versionUpdated, typeOk := updated.(int64)
```

Comment:
I think this will cause nil pointer exception? We should probably check ok first

Reply:
It doesn't seem to cause nil pointer `DEBUG PRUNE INSIDE updated <nil> ok false versionUpdated 0 typeOk false`. So even if the syncMap doesn't contain an entry for `storeKey` its fine to do type assertion on it.

Verifying this again